### ISSUE TITLE
Cannot format unsaved code

### ISSUE THREAD

**Issue by j-ulrich** [xaverh/vscode-clang-format-provider#110]
If I try to format code which has not been saved to a file on disk yet, I get the message
```
There is no formatter for 'cpp' files installed.
```

I have the setting `clang-format.assumeFilename` configured and it works correctly as soon as the file is saved anywhere on disk.

It also doesn't matter if I have the setting `C_Cpp.formatting` disabled or set to "Default".

I'm running on macOS with zsh as the default shell, in case it matters.

**Comment by bdeeming** [comment]
Have you tried this to force the choice?
```
"[cpp]": {
    "editor.defaultFormatter": "xaver.clang-format"
  }
```

**Comment by j-ulrich** [comment]
> Have you tried this to force the choice?
> 
> ```
> "[cpp]": {
>     "editor.defaultFormatter": "xaver.clang-format"
>   }
> ```

Yes, this is set in my `settings.json` but the message still appears.
